tools/lib/NQP/Config/Rakudo.pm

```diff
@@ -475,6 +475,38 @@ sub opts_for_configure {
     return wantarray ? @opts : join( " ", @opts );
 }
 
+sub clean_old_p6_libs {
+    my $self = shift;
+    my $is_moar  = $self->active_backend('moar');
+    if ( $is_moar ) {
+        my $nqp_config = $self->{impls}{moar}->{config};
+        my $lib_dir = File::Spec->rel2abs(File::Spec->catdir( $nqp_config->{'nqp::prefix'}, 'share', 'nqp', 'lib', 'Perl6' ));
+
+        return if !-d $lib_dir;
+
+        my @files = qw(Actions.moarvm BOOTSTRAP.moarvm Compiler.moarvm Grammar.moarvm Metamodel.moarvm ModuleLoader.moarvm Ops.moarvm Optimizer.moarvm Pod.moarvm World.moarvm);
+
+        my $said;
+        for ( @files ) {
+            my $file = File::Spec->catdir( $lib_dir, $_ );
+            next unless -f $file;
+            if ( !$said ) {
+                $self->msg("\n");
```

Comment:
There is method `note` for uniform important messages. The whole bunch of `->msg()` calls should be replaced with:

```
$self->note('NOTICE', @msg_lines);
```

Besides, I would only collect files within the loop. And then when it's over check if anything was collected then print a note which would include the list of files. But that just a suggestion, nothing serious.

Reply:
Done.